src/palace/manager/api/sirsidynix_authentication_provider.py

```diff
@@ -194,22 +193,30 @@ def remote_authenticate(
 
     def remote_patron_lookup(
         self, patron_or_patrondata: Patron | PatronData
-    ) -> None | SirsiDynixPatronData:
+    ) -> None | SirsiDynixPatronData | PatronData:
         """Do a remote patron lookup, this method can only look up a patron with a patrondata object
         with a session_token already setup within it.
         This method also checks all the reasons that a patron may be blocked for.
         """
-        # We cannot do a remote lookup without a session token
-        if not isinstance(patron_or_patrondata, SirsiDynixPatronData):
-            return None
-        elif not patron_or_patrondata.session_token:
+
+        # Return None if a patron object is passed in.
+        if isinstance(patron_or_patrondata, Patron):
             return None
 
+        # if the patron data object is not authenticated just pass it back after ensuring the
+        # complete flag is set to False
+        if (
+            not hasattr(patron_or_patrondata, "session_token")
+            or patron_or_patrondata.session_token is None
+        ):
```

Comment:
You can eliminate all the `# type: ignore` comments below by updating this to be:
```python
  if (
      not isinstance(patron_or_patrondata, SirsiDynixPatronData)
      or patron_or_patrondata.session_token is None
  ):
      patron_or_patrondata.complete = False
      return patron_or_patrondata
```